app/todo/repository.py

```diff
@@ -0,0 +1,62 @@
+import abc
+
+from sqlalchemy import select, delete
+from sqlalchemy.ext.asyncio import AsyncSession
+from app.todo.models import TodoInfo
+
+from app.db import tables as tb
+
+
+class BaseTodoRepository(abc.ABC):
+    def __init__(self, session: AsyncSession) -> None:
+        self.session = session
+
+    @abc.abstractmethod
+    async def getTodoInfoList(self, user_id: int) -> list[TodoInfo]:
+        raise NotImplementedError
+    
+    @abc.abstractmethod
+    async def insertTodoInfo(self, user_id: int, context: str) -> TodoInfo:
+        raise NotImplementedError
+    
+    @abc.abstractmethod
+    async def deleteTodoInfo(self, user_id: int, todo_id: int) -> int:
+        raise NotImplementedError
+
+
+class TodoRepository(BaseTodoRepository):
+    async def getTodoInfoList(self, user_id: int) -> list[TodoInfo]:
+        results = (
+            (
+                await self.session.execute(
+                    select(tb.Todo)
+                    .filter(
+                        tb.Todo.user_id == user_id
+                    )
+                )
+            )
+            .unique()
+            .scalars()
+            .all()
+        )
+        return [TodoInfo.model_validate(row) for row in results]
+    
+    async def insertTodoInfo(self, user_id: int, context: str) -> TodoInfo:
+        new_todo = tb.Todo(user_id=user_id, context=context)
+        self.session.add(new_todo)
+        await self.session.commit()
+
+        return TodoInfo(
+            todo_id = new_todo.todo_id,
+            user_id = new_todo.user_id,
+            context = new_todo.context,
+            created_at = new_todo.created_at,
+            modified_at = new_todo.modified_at,
+        )
+
+    async def deleteTodoInfo(self, user_id: int, todo_id: int) -> str:
+        result = await self.session.execute(delete(tb.Todo).where((tb.Todo.todo_id == todo_id) & (tb.Todo.user_id == user_id)))

```

Comment:
```suggestion
        result = await self.session.execute(delete(tb.Todo).where(tb.Todo.todo_id == todo_id))

```

todo_id가 이미 unique하므로 user_id에 대한 condition이 없어도 의도된대로 동작할 것 같습니다! (p3)

Reply:
혹시 모를 보안을 위해 해당 조건절을 추가했습니다.
만약 ROW의 user_id가 1일 때, user_id 가 1인 유저가 todo_id를 바꿔서 진행하면 해당 로직이 실행되기에
추가했습니다.